waku/v2/protocol/relay/broadcast.go

```diff
@@ -62,36 +65,40 @@ func (s *chStore) broadcast(ctx context.Context, m *protocol.Envelope) {
 		// this will also block the chStore close function as it uses same mutex
 		case <-ctx.Done():
 			return
-		case ch <- m:
+		default:
+			sub.Submit(m)
 		}
 	}
-	// send to all registered subscribers
-	for _, ch := range s.topicToChans[""] {
+
+	// send to all wildcard subscribers
+	for _, sub := range s.topicsToSubs[""] {
 		select {
 		case <-ctx.Done():
 			return
-		case ch <- m:
+		default:
+			sub.Submit(m)
```

Comment:
This `Submit` function can blocking, since ctx.Done() would not really cancel a submission. One way to solve this could be by changing the function so it looks like this:

```go
// Submit allows a message to be submitted for a subscription
func (s *Subscription) Submit(ctx context.Context, msg *protocol.Envelope) {
	//Filter and notify
	// - if contentFilter doesn't have a contentTopic
	// - if contentFilter has contentTopics and it matches with message
	if len(s.contentFilter.ContentTopicsList()) == 0 || (len(s.contentFilter.ContentTopicsList()) > 0 &&
		slices.Contains[string](s.contentFilter.ContentTopicsList(), msg.Message().ContentTopic)) {
                select {
                    case <-ctx.Done():
			return
                    case s.Ch <- msg:
                        //
                }
	}
}
```

Notice the `select` block i added. That way if the context is canceled, the attempt to push the `msg` to `s.Ch` will not stop the execution of `Submit`